x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -288,11 +289,16 @@ private void startComputeOnDataNodes(
         ActionListener<Void> parentListener,
         Supplier<ActionListener<ComputeResponse>> dataNodeListenerSupplier
     ) {
+        var planWithReducer = dataNodePlan.transformUp(FragmentExec.class, f -> {
+            PhysicalPlan reductionNode = PlannerUtils.dataNodeReductionPlan(f.fragment(), dataNodePlan);
```

Comment:
Determine the reducer physicalplan node only once, on the coordinator node. Attach it to the fragment and send it to data nodes.